File: AlgoPersonalStudy/src/backjoon/silver/BJ_1697.java
package backjoon.silver;

import java.io.*;
import java.util.*;

public class BJ_1697 {
    static int MAX_LEN = 100001;
    static int[] delta = {-1,1};
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());
        if(n == k){
            System.out.println(0);
            return;
        }

        Queue<int[]> queue = new LinkedList<>();
        int[] map = new int[MAX_LEN];
        Arrays.fill(map, Integer.MAX_VALUE);
        int day = 0;

        queue.add(new int[]{0,n});
        while(!queue.isEmpty()){
            day++;
            int[] now = queue.poll();
            int dx = now[1] + 1;
            if(dx < MAX_LEN){
                if(day < map[dx]){
                    map[dx] = day;
                    queue.add(new int[]{day, dx});
                }
            }
            dx = now[1] - 1;
            if(dx >= 0){
                if(day < map[dx]){
                    map[dx] = day;
                    queue.add(new int[]{day, dx});
                }
            }
            dx = now[1] * 2;
            if(dx < MAX_LEN){
                if(day < map[dx]){
                    map[dx] = day;
                    queue.add(new int[]{day, dx});
                }
            }
        }

        System.out.println(map[k]);
    }
}

File: AlgoPersonalStudyPython/backjoon/silver/BJ_1697.py
import sys
from queue import Queue


def main():
    max_len = 100001
    val_n, val_k = map(int, sys.stdin.readline().split())
    if val_n == val_k:
        print(0)
        return
    qu = Queue()
    map_int = []
    day = 0
    qu.put([0, val_n])
    while len(map_int) > 0:
        day += 1
        now = qu.get()
        dx = now[1] + 1
        if dx < max_len:
            if day < map_int[dx]:
                map_int[dx] = day
                qu.put([day, dx])
        dx = now[1] - 1
        if dx >= 0 and day < map_int[dx]:
            map_int[dx] = day
            qu.put([day, dx])
        dx = now[1] * 2
        if dx < max_len and day < map_int[dx]:
            map_int[dx] = day
            qu.put([day, dx])
    print(map_int[val_k])


if __name__ == "__main__":
    main()
